File: API/Entity/Transaction.cs
using System;

namespace API.Entity;

public class Transaction
{
    public int Id { get; set; }
    public int UserId { get; set; }
    public AppUser? User { get; set; }
    public required string Description { get; set; }
    public decimal Amount { get; set; }
    public DateTime Date { get; set; }
    public Category? Category { get; set; }
    public int CategoryId { get; set; }

}

File: frontend/src/app/models/transaction.ts
export type Transaction = {
    id: number
    userId: number
    description: string
    amount: number
    date: Date
    categoryId: number
}

export type TransactionUpdateModel = {
    id: number
    description: string
    amount: number
    date: Date
    categoryId: number
}

export type TransactionParams = {
    orderby: string,
    searchTerm: string,
    categories: string[],
    dateFrom: string,
    dateTo: string

}